Ardagryd.js

```diff
@@ -14,6 +14,51 @@ const DESCENDING = "desc";
 //Workaround. See -> https://phabricator.babeljs.io/T6777
 typeof undefined;
 
+
+
+//helpers
+
+let sortConfigFromProp = (sortProp) => {
+    let sortColumn;
+    let order = ASCENDING;
+
+    switch (typeof  sortProp){
+        case "string":
+            sortColumn = sortProp;
+            break;
+        case "object":
+            let sortConfigObject;
+            if (Array.isArray(sortProp) && sortProp.length > 0){
+                sortConfigObject = sortProp[0];
+            } else {
+                sortConfigObject = sortProp;
+            }
+            sortColumn = sortConfigObject.columnName;
+            if (sortConfigObject.order
+                && (sortConfigObject.order === ASCENDING || sortConfigObject.order === DESCENDING)){
+                order = sortConfigObject.order;
+            }
+            break;
+        default:
```

Comment:
When the `default` branch executes, i.e. if `sortProp` is neither a string nor an object/array. We could also just let is break in production. That's going to save us a few milliseconds and `propTypes` should have made sure that the user passes in valid data. Let's see if it causes trouble for anyone, we can also act on this later.